### ISSUE TITLE
HAProxy deletes Loadbalancer field of other Ingresses 

### ISSUE THREAD

**Issue by Antaxify** [haproxytech/helm-charts#198]
**Our Setup:**
- nginx ingress controller with a default ingress class, used as regular reverse proxy
- haproxy (installed with this chart) with a non-default ingress class, used to route L4 traffic (--configmap-tcp-services)

After upgrading from 1.30.5 to 1.32.0, we noticed that the Loadbalancer IPs (status.loadBalancer.ingress) of our regular ingress resources disappearing, which in turn triggered external-dns to delete the corresponding DNS records. After rolling back to 1.30.5, the issue was resovled.

**Steps taken:**
- double checked our config, everything seems fine
- ran helm template with our config for both 1.30.5 and 1.32.0, only the haproxy version changed (patch release)
- deployed 1.32.0 one more time just to reproduce the issue and to be sure haproxy is the cause

**Helm Values:**
```
podSecurityPolicy:
  enabled: false
rbac:
  create: true
namespace:
  create: false
serviceAccount:
  create: true
controller:
  name: controller
  image:
    repository: haproxytech/kubernetes-ingress
    pullPolicy: IfNotPresent
  kind: Deployment    # can be 'Deployment' or 'DaemonSet'
  replicaCount: 2
  minReadySeconds: 0
  unprivileged: true
  enableRuntimeDefaultSeccompProfile: true
  terminationGracePeriodSeconds: 60
  extraArgs:
  - --configmap-tcp-services=ingress-haproxy/tcp-services
  containerPort:
    http: 8080
    https: 8443
    stat: 1024
  livenessProbe:
    failureThreshold: 3
    initialDelaySeconds: 0
    periodSeconds: 10
    successThreshold: 1
    timeoutSeconds: 1
    httpGet:
      path: /healthz
      port: 1042
      scheme: HTTP
  readinessProbe:
    failureThreshold: 3
    initialDelaySeconds: 0
    periodSeconds: 10
    successThreshold: 1
    timeoutSeconds: 1
    httpGet:
      path: /healthz
      port: 1042
      scheme: HTTP
  startupProbe:
    failureThreshold: 20
    initialDelaySeconds: 0
    periodSeconds: 1
    successThreshold: 1
    timeoutSeconds: 1
    httpGet:
      path: /healthz
      port: 1042
      scheme: HTTP
  ingressClassResource:
    name: haproxy
    default: false
  ingressClass: haproxy   # typically "haproxy" or null to receive all events
  enableServiceLinks: true
  defaultTLSSecret:
    enabled: true
  resources:
    requests:
      cpu: 250m
      memory: 400Mi
  autoscaling:
    enabled: false
  nodeSelector:
    persistent: "true"
  config:
  #   timeout-connect: "250ms"
  #   servers-increment: "10"
  #   servers-increment-max-disabled: "10"
  #   rate-limit: "ON"
  #   rate-limit-expire: "1m"
  #   rate-limit-interval: "10s"
  #   rate-limit-size: "100k"
  configAnnotations: {}
  keda:
    enabled: true
    minReplicas: 2
    maxReplicas: 10
    pollingInterval: 30
    cooldownPeriod: 300
    restoreToOriginalReplicaCount: true
    behavior:
      scaleDown:
        stabilizationWindowSeconds: 300
        policies:
        - type: Percent
          value: 100
          periodSeconds: 15
    triggers:
    - type: prometheus
      metadata:
        customHeaders: some-header
        serverAddress: some-url
        metricName: haproxy_process_idle_time_percent
        threshold: '50'
        query: avg(100-avg_over_time(haproxy_process_idle_time_percent{service="haproxy-kubernetes-ingress"}[2m]))
  logging:
    level: info
  publishService:
    enabled: true
    pathOverride: ""
  service:
    enabled: true     # set to false when controller.kind is 'DaemonSet' and controller.daemonset.useHostPorts is true
    type: LoadBalancer    # can be 'ClusterIP', 'NodePort' or 'LoadBalancer'
    annotations:
      networking.gke.io/load-balancer-type: Internal
    ports:
      http: 80
      https: 443
      stat: 1024
    enablePorts:
      http: true
      https: true
      stat: true
    targetPorts:
      http: http
      https: https
      stat: stat
    tcpPorts:
    - name: otlp
      port: 4317
      targetPort: 4317
    - name: zipkin
      port: 9411
      targetPort: 9411
    externalTrafficPolicy: "Local"
    loadBalancerIP: someIP
    loadBalancerSourceRanges: ["10.0.0.0/8"]
  serviceMonitor:
    enabled: false
    extraLabels: {}
    endpoints:
    - port: stat
      path: /metrics
      scheme: http
      interval: 30s
```

**Comment by dkorunic** [comment]
Hi @Antaxify, thank you for a very thorough bug report. I've rendered Chart with your overrides exactly as above and compared output between `1.30.5` and `1.32.0` versions; I am quite sure the bug is not Chart related, since the differences at last Chart wise are trivial and unrelated to the problem described above. I'll proceed to check this issue with IC development team and see what we can find.

**Comment by dkorunic** [comment]
Fix is in the https://github.com/haproxytech/kubernetes-ingress/commit/a3368dd56be0ca9f1f8fd19ba64d59ea48e08a63 and it has been merged and released in IC [v1.10.6](https://github.com/haproxytech/kubernetes-ingress/releases/tag/v1.10.6)
